nengo/conftest.py

```diff
@@ -210,12 +231,17 @@ def pytest_runtest_setup(item):
                 ('logger', 'logs', "logs not requested")]:
             if fixture_name in item.fixturenames:
                 if item.config.getvalue(option):
-                    skip = False
+                    break
                 else:
                     skipreasons.append(message)
-        if skip:
+        else:
             pytest.skip(" and ".join(skipreasons))
 
+    if 'Simulator' in item.fixturenames:
+        for test, reason in _Simulator.unsupported:
```

Comment:
The answer to this might also depend on how long do we expect the list of unsupported features to be?

But as long as we stick to regexes we probably have to match each regex against each test name.

One minor optimization could be to compile all of the regexes in a first pass. At the moment we compile each regex for each test.